backend/server/src/main/java/ai/verta/modeldb/entities/DatasetEntity.java

```diff
@@ -243,7 +244,12 @@ public Dataset getProtoObject(MDBRoleService mdbRoleService) {
             ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET);
     datasetBuilder.setVisibility(repositoryResource.getVisibility());
     datasetBuilder.setWorkspaceServiceId(repositoryResource.getWorkspaceId());
-    datasetBuilder.setOwner(String.valueOf(repositoryResource.getOwnerId()));
+    if (repositoryResource.getOwnerTrackingCase() == OwnerTrackingCase.GROUP_OWNER_ID) {
+      datasetBuilder.setGroupOwnerId(repositoryResource.getGroupOwnerId());
```

Comment:
is `datasetBuilder.setOwner()` not required for the `if` block?

Reply:
In case of if condition is true we will have owner set to default value ("")

Reply:
I think now I got what you mean: we should set default value because datasetBuilder.setOwner may be already called before? Makes sense, added setOwner("")